uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/manage-leave-requests.spec.js

```diff
@@ -1,929 +1,953 @@
+      spyOn(AbsenceType, 'all').and.callThrough();
+      spyOn(Contact, 'leaveManagees').and.callFake(function () {
+        return ContactAPIMock.leaveManagees();
       });
-
-      it('is initialized', function () {
-        expect($log.debug).toHaveBeenCalled();
+      spyOn(Contact, 'all').and.callFake(function (filters) {
+        return ContactAPIMock.all(filters);
       });
+      spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+        return OptionGroupAPIMock.valuesOf(name);
+      });
+    });
 
-      describe('initially', function () {
-        it('the filter section is closed', function () {
-          expect(controller.isFilterExpanded).toBe(false);
-        });
+    beforeEach(function () {
+      compileComponent();
+    });
 
-        describe('pagination', function () {
-          it('is set to page 1', function () {
-            expect(controller.pagination.page).toBe(1);
-          });
+    it('is initialized', function () {
+      expect($log.debug).toHaveBeenCalled();
+    });
 
-          it('has page size set to 7', function () {
-            expect(controller.pagination.size).toBe(7);
-          });
-        });
+    describe('initially', function () {
+      it('the filter section is closed', function () {
+        expect(controller.isFilterExpanded).toBe(false);
       });
 
-      describe('data loading', function () {
-        // TODO need to figure out how to test variables which are changed when controller gets initialized
-        describe('before loading starts', function () {
-          it('loader is hidden', function () {});
-          it('leave requests are empty', function () {});
-          it('absencePeriods are empty', function () {});
-          it('absenceTypes are empty', function () {});
-          it('statusCount is reset', function () {});
+      describe('pagination', function () {
+        it('is set to page 1', function () {
+          expect(controller.pagination.page).toBe(1);
         });
 
-        describe('after data loading is complete', function () {
-          describe('loading', function () {
-            it('loader is hidden for page', function () {
-              expect(controller.loading.page).toBe(false);
-            });
-
-            it('loader is hidden for content', function () {
-              expect(controller.loading.content).toBe(false);
-            });
-          });
+        it('has page size set to 7', function () {
+          expect(controller.pagination.size).toBe(7);
+        });
+      });
+    });
 
-          it('leave requests status have loaded', function () {
-            var expectedResult = optionGroupMock.getCollection('hrleaveandabsences_leave_request_status').concat({
-              name: 'all',
-              label: 'All'
-            });
+    describe('data loading', function () {
+      // TODO need to figure out how to test variables which are changed when controller gets initialized
```

Comment:
Hint: use the `$onInit` method! Not sure how to apply it for this case though.

But the gist is to use the constructor to setup the controller before initializing it, you test these settings on your spec file, then run `$onInit` manually and test the initialization effects.

Note: this is just a comment.